src/__lib__/html.js

```diff
@@ -36,10 +37,24 @@ export default ({
             <script>
             window.gu = ${JSON.stringify({
                 app: {
-                    data,
+                    data: {
+                        ...data,
+                        content: {
+                            ...data.content,
+                            ...contentReferences.reduce(
+                                (acc, contentReference) => {
+                                    acc[
+                                        contentReference
+                                    ] = `document.querySelector('[data-content-${contentReference}]').innerHTML`;
+                                    return acc;
+                                },
+                                {},
+                            ),
+                        },
+                    },
                     cssIDs,
                 },
-            })};
+            }).replace(/"(document.*?innerHTML)"/g, '$1')};
```

Comment:
JS executing on the server to generate JS that is passed to the client and executed to retrieve data that was added to HTML that was generated on the server. `:mindblown:`

Reply:
ha! maybe we could write a `sanitiseDomRefs` method or something, so it's a bit clearer?